### ISSUE TITLE
The BroadcasterFuture has a bug in isDone() method -> it sets done to true

### ISSUE THREAD

**Issue by wkozaczuk** [Atmosphere/atmosphere#2088]
```java
    @Override
    public boolean isDone() {

        if (innerFuture != null) {
            return innerFuture.isDone();
        }

        isDone = true;
        return isDone;
    }

```

**Comment by wkozaczuk** [comment]
Also I don't this it works if latchCount > 1 if internal future is null:
     */
    public BroadcasterFuture<E> done() {
        isDone = true;

        if (latch != null) {
            latch.countDown();
        }
        return this;
    }

**Comment by jfarcand** [comment]
@wkozaczuk Pull request welcomed :-)

**Comment by rcamposg** [comment]
Hello,

We are facing the same issue when we are using BroadcasterFuture.get() to know when the broadcaster has finished delivering the messages.

As a workaround, we are using the alternative with timeout, so the thread is not hung forever.

**Comment by elakito** [comment]
@wkozaczuk @rcamposg I just committed the fix to master, 2.4.x, and 2.3.x.
regards, aki
